api/v1alpha1/worker_types.go

```diff
@@ -105,19 +105,14 @@ type TemporalWorkerDeploymentStatus struct {
 	// TargetVersion is the desired next version. If TargetVersion.Deployment is nil,
 	// then the controller should create it. If not nil, the controller should
 	// wait for it to become healthy and then move it to the CurrentVersion.
+	// This must never be nil.
 	TargetVersion *TargetWorkerDeploymentVersion `json:"targetVersion"`
```

Comment:
```suggestion
TargetVersion TargetWorkerDeploymentVersion `json:"targetVersion"`
```

If it's expected that it can't be nil, what do you think about enforcing that in the API?

Reply:
Yes I think we should, what did you have in mind?

Reply:
I've added some validation around this.

Reply:
I meant to remove the pointer value (see suggested change) -- I think that should make it required in the json schema and also avoid the need for nil checks in the controller code.